src/base/utils/string.cpp

```diff
@@ -144,30 +172,43 @@ namespace
 
 bool Utils::String::naturalCompareCaseSensitive(const QString &left, const QString &right)
 {
-    // provide a single `NaturalCompare` instance for easy use
-    // https://doc.qt.io/qt-5/threads-reentrancy.html
-#ifdef Q_OS_MAC  // workaround for Apple xcode: https://stackoverflow.com/a/29929949
-    static QThreadStorage<NaturalCompare> nCmp;
-    if (!nCmp.hasLocalData()) nCmp.setLocalData(NaturalCompare(true));
-    return (nCmp.localData())(left, right);
-#else
-    thread_local NaturalCompare nCmp(true);
-    return nCmp(left, right);
-#endif
+    return NaturalCompare::naturalCompareCaseSensitive(left, right) < 0;
```

Comment:
You don't really have to create this function in order to let `Utils::String::naturalCompareCaseSemiSensitive` looks nice... or you could just propagate the return int up as I done here: https://github.com/Chocobo1/qBittorrent/commit/20964cef218db7406c41af3281cffa0969d0f843
Just some ideas though...